### PR TITLE
Document how to server WebDAV and directory listing under same path.

### LINKED ISSUE #20: Is it possible to have directory listing with WebDAV enabled in the same path?
Based on the documents and my test, it seems like WebDAV has to be served in a different path with directory listing, one cannot have them both functioning under the same path, is that correct?

As an Apache & Nginx user, I felt quite surprised, as it is an expected behaviour to have them working in the same path.

Cheers,
Jayden

### PR DESCRIPTION
**Pull request by rekacant**
Thanks for help from @wader, @Zeethulhu and @noerw.

This should close #20.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,6 +51,38 @@ webdav /some/path/match/* {
 }
 ```
 
+If you want to serve WebDAV and directory listing under same path, e.g. similer behaviour as in Apache and Nginx, you may use [Request Matchers](https://caddyserver.com/docs/caddyfile/matchers) to filter out GET request and pass it to [file_server](https://caddyserver.com/docs/caddyfile/directives/file_server).
+
+Example with authenticated WebDAV and directory listing under the same path:
+
+```
+@notget {
+    not method GET
```
Comment (mholt) [inline README.md:60]: Any special reason that `not method GET` is used instead of `method GET` (and then applying it to the `file_server` instead)?

**Review by mholt — CHANGES_REQUESTED** [review]
Thanks for the suggested change. Just a couple items of feedback.

File: README.md [hunk 2]
```diff
@@ -51,29 +55,32 @@ webdav /some/path/match/* {
...
-    webdav @notget
-    file_server browse
+    file_server @get browse
+    webdav
 }
 ```
 
-Or, if you want to create a public listing, but keep WebDAV behind authentication:
+Or, if you want to create a public listing, but keep WebDAV behind
+authentication:
```
Comment (mholt) [inline README.md:79]: What's with the arbitrary line breaks?

Please clean up the diff and then this LGTM. :+1:

File: README.md (hunk 1 above)
Comment (rekacant) [reply README.md]: Just personal preference, no special reason.

I will leaving both "get" and "not get" macher here for demonstration purpose.

File: README.md (hunk 2 above)
Comment (rekacant) [reply README.md]: Sorry 🙇🏽‍♂️

**Review by mholt — APPROVED** [review]
LGTM, thanks!

I wonder if passing GET requests through webdav should be the default behavior so this extra config isn't necessary.

**Pull request merged**